utils/utils.go

```diff
@@ -53,7 +53,16 @@ func HostZoneUnbondingKeys(m map[string]*recordstypes.HostZoneUnbonding) []strin
 	return keys
 }
 
-func StringToIntMapKeys(m map[string]int64) []string {
+func StringToSdkIntMapKeys(m map[string]sdk.Int) []string {
```

Comment:
I think we can probably generalize all these with:
```go
func StringMapKeys(m map[string]interface{}) []string {
    keys := make([]string, 0, len(m))
    for k := range m {
	keys = append(keys, k)
    }
    sort.Strings(keys)
    return keys
}
```

Then we wouldn't need `StringToSdkIntMapKeys`, `StringToSdkDecMapKeys`,`StringToStringMapKeys`,  and `StringToStringSliceMapKeys`

Reply:
I tried this when we first wrote these functions and it didn't work, will give it another look

Reply:
```
cannot use validatorDeltas (variable of type map[string]"github.com/cosmos/cosmos-sdk/types".Int) as map[string]interface{} value in argument to utils.StringMapKeyscompiler[IncompatibleAssign](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal?utm_source%3Dgopls#IncompatibleAssign)
```
here's an example of the error

Reply:
Something like this might work
```
func StringMapKeys[V any](m map[string]V) []string {
    keys := make([]string, 0, len(m))
    for k := range m {
	keys = append(keys, k)
    }
    sort.Strings(keys)
    return keys
}
```

Reply:
You might just have to cast it before passing it in?

```go
StringMapKeys(validatorDeltas.(map[string]interface{}))
```

Reply:
I think the above approach using generics is slightly cleaner, and seems to work! https://go.googlesource.com/proposal/+/refs/heads/master/design/43651-type-parameters.md#map-keys

Reply:
Agreed, looks good to me!